internal/client/client.go

```diff
@@ -10,14 +11,38 @@ import (
 	"github.com/google/uuid"
 
 	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+
+	retryablehttp "github.com/hashicorp/go-retryablehttp"
 )
 
 var _ = api.PrefectClient(&Client{})
 
 // New creates and returns new client instance.
 func New(opts ...Option) (*Client, error) {
+	// Uses the retryablehttp package for built-in retries
+	// with exponential backoff.
+	//
+	// Some notable defaults from that package include:
+	// - max retries: 4
+	// - retry wait minimum seconds: 1
+	// - retry wait maximum seconds: 30
+	//
+	// All defaults are defined in
+	// https://github.com/hashicorp/go-retryablehttp/blob/main/client.go#L48-L51.
+	retryableClient := retryablehttp.NewClient()
+
+	// By default, retryablehttp will only retry requests if there was some kind
+	// of transient server or networking error. We can be more specific with this
+	// by providing a custom function for determining whether or not to retry.
+	retryableClient.CheckRetry = checkRetryPolicy
+
+	// Finally, convert the retryablehttp client to a standard http client.
+	// This allows us to retain the `http.Client` interface, and avoid specifying
+	// the `retryablehttp.Client` interface in our client methods.
+	httpClient := retryableClient.StandardClient()
+
 	client := &Client{
-		hc: http.DefaultClient,
+		hc: httpClient,
```

Comment:
This is the main change here - we're replacing the default HTTP client with our retry-able HTTP client.